Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceUpdateExecutor.cpp

```diff
@@ -94,9 +121,108 @@ namespace AzToolsFramework
+            
+            PrefabDomPath domSourcePath(aliasPath.c_str());
+            PrefabDom partialInstanceDom;
+            partialInstanceDom.CopyFrom(
+                m_prefabSystemComponentInterface->FindTemplateDom(domSourceInstance->GetTemplateId()), instanceDom.GetAllocator());
+
+            auto instanceDomValueFromSource = domSourcePath.Get(partialInstanceDom);
+            if (!instanceDomValueFromSource)
+            {
+                return false;
+            }
+
+            instanceDom.CopyFrom(*instanceDomValueFromSource, instanceDom.GetAllocator());
+
+            // If the focused instance is not an ancestor of our instance, verify if it's a descendant.
+            if (domSourceInstance != &focusedInstance->get())
+            {
+                AZStd::string aliasPathToFocus;
+                auto focusedInstanceAncestor = PrefabDomUtils::ClimbUpToTargetInstance(&focusedInstance->get(), instance, aliasPathToFocus);
+
+                // If the focused instance is a descendant (or the instance itself), we need to replace its portion of the dom with the template one.
+                if (focusedInstanceAncestor == instance)
+                {
+                    // Get the dom for the focused instance from its template.
+                    PrefabDom focusedInstanceDom;
+                    focusedInstanceDom.CopyFrom(
+                        m_prefabSystemComponentInterface->FindTemplateDom(focusedInstance->get().GetTemplateId()),
+                        instanceDom.GetAllocator());
+
+                    // Replace the container entity with the one as seen by the root
+                    // TODO - this function should only replace the transform!
+                    ReplaceFocusedContainerTransformAccordingToRoot(&focusedInstance->get(), focusedInstanceDom);
+                    
+                    // Copy the focused instance dom inside the dom that will be used to refresh the instance.
+                    PrefabDomPath domSourceToFocusPath(aliasPathToFocus.c_str());
+                    domSourceToFocusPath.Set(instanceDom, focusedInstanceDom, instanceDom.GetAllocator());
+
+                    // Force a deep copy
+                    PrefabDom instanceDomCopy;
+                    instanceDomCopy.CopyFrom(instanceDom, instanceDom.GetAllocator());
+
+                    instanceDom.CopyFrom(instanceDomCopy, instanceDom.GetAllocator());
+                }
+            }
+            // If our instance is the focused instance, fix the container
+            else if (&focusedInstance->get() == instance)
             {
-                return entry == instance;
-            });
+                // Replace the container entity with the one as seen by the root
+                // TODO - this function should only replace the transform!
+                ReplaceFocusedContainerTransformAccordingToRoot(instance, instanceDom);
```

Comment:
Is this change to ReplaceFocusedContainerTransformAccordingToRoot planned as a task?